aiida_pseudo/groups/mixins/cutoffs.py

```diff
@@ -90,60 +91,92 @@ def validate_stringency(self, stringency: str) -> None:
         if stringency not in self.get_cutoff_stringencies():
             raise ValueError(f'stringency `{stringency}` is not defined for this family.')
 
-    def _get_cutoffs(self) -> dict:
-        """Return the cutoffs dictionary.
+    def _get_cutoffs_dict(self) -> dict:
+        """Return the cutoffs dictionary that maps the stringencies to the recommended cutoffs.
 
         :return: the cutoffs extra or an empty dictionary if it has not yet been set.
         """
         return self.get_extra(self._key_cutoffs, {})
 
+    def _get_cutoffs_unit_dict(self) -> str:
+        """Return the cutoffs units for each of the stringencies.
+
+        :return: the cutoffs units extra or an empty dictionary if it has not yet been set.
+        """
+        return self.get_extra(self._key_cutoffs_unit, {})
+
     def get_default_stringency(self) -> Union[str, None]:
         """Return the default stringency if defined.
 
         :return: the default stringency or ``None`` if no cutoffs have been defined for this family.
-        :raises ValueError: if not default stringency has been defined.
+        :raises ValueError: if default stringency has not been defined.
```

Comment:
I noticed that the current docstring and return type of `get_default_stringency` is incorrect. It says it returns `None` if not defined but it actually raises. Maybe raising is actually the best behavior? If we have it return `None`, we should maybe change the implementation of `get_cutoffs_unit` which now assumes it gets a value back

Reply:
I guess it depends on what error we want the user to see when they try to use `get` methods and have no stringencies defined. Once _any_ stringency is defined with the `set_cutoffs` method, it will automatically be set to the default. I think raising with an error message like `'no recommended cutoffs have been specified for this pseudo family'` might be better than `'no default stringency has been defined.'` when no stringencies have been defined _at all_.

However, now that I think of it, we should probably also add a `remove_cutoffs` method, since now there is no API for doing so. If this method is implemented, it's possible that the user:

1. Uses `set_cutoffs` to set the `'normal'` stringency, which becomes the default.
2. Uses `set_cutoffs` to set to `'high'` stringency.
3. Removes the `'normal'` stringency.

What do we do in this instance? Since there is only one stringency left, it makes sense to make that the default, but what if there are 3 stringencies at some point, and the default is removed? Do we let the user simply not remove the default stringency (raise), or allow stringencies/cutoffs to be defined without having a default set?

Reply:
Good points. Seems like that really we should just accept that it is very much possible to not have a default stringency defined, which means that we should simply have it raise when it is asked for. I think that `'no default stringency has been defined'` in this case is then better than `'no recommended cutoffs have been specified for this pseudo family' ` because not only does the latter tie an implementation detail into error message, but once we add the `delete_cutoffs` method, that won't necessarily be true anymore.

TLDR: my vote
* let `get_default_stringency` raise if not defined
* `set_cutoffs` automatically sets default if only one stringency is defined
* implement `delete_cutoffs` that will also remove the default stringency if that corresponds (only sets new default if only stringency remains)